openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TableRepository.java

```diff
@@ -558,6 +558,21 @@ public Table deleteCustomMetric(UUID tableId, String columnName, String metricNa
 
   public Table addDataModel(UUID tableId, DataModel dataModel) {
     Table table = dao.findEntityById(tableId);
+
+    // Update the sql fields only if correct value is present
+    if (dataModel.getRawSql() == null || dataModel.getRawSql().isBlank()) {
```

Comment:
`null` or `blank` sql and rawsql will not overwrite already present sql with this logic added.